charts/elasticsearch/templates/master/es-master-service.yaml

```diff
@@ -4,15 +4,19 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "elasticsearch.fullname" . }}-discovery
+  name: {{ template "elasticsearch.fullname" . }}-headless-discovery
   labels:
     tier: logging
     component: {{ template "elasticsearch.name" . }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
     role: master
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
```

Comment:
depending on the prometheus config it may never be able to scrape metrics from the "service". By making it headless there is not an IP for prometheus to hit based on the service. It would have to know to scrape the actual endpoints referenced via the service, rather than the service IP itself. Again all depends on prometheus is configured wether this will work or not.

Reply:
We don't have a metrics endpoint being scraped on the master service. We are using an exporter that scrapes from the ES ingress nodes' (we are calling the ingress nodes 'clients') service:

https://github.com/astronomer/astronomer/blob/113b61f43be428faf01a68788fcbafa4aff35ad6/charts/elasticsearch/templates/exporter/es-exporter-deployment.yaml#L58-L64

https://github.com/astronomer/astronomer/blob/113b61f43be428faf01a68788fcbafa4aff35ad6/charts/elasticsearch/templates/client/es-client-service.yaml#L7

Reply:
I don't know if this is the best way to do it, it does seem like a lot of hops for a metrics scrape. I think it would go like this on every scrape.

prom > exporter > client > master (then all the way back)